whylog/assistant/regex_assistant/regex.py

```diff
@@ -0,0 +1,85 @@
+"""
+Regex verification and creating (but not finding groups in regex)
+"""
+
+import re
+
+from whylog.assistant.regex_assistant.exceptions import NotMatchingRegexError
+
+special_characters = frozenset('.^$*+{}?[]|()]')
+
+
+def verify_regex(regex, text):
+    """
+    Verifies regex properties such as:
+    - matching a whole text
+    - matching text in a one way only
+    If properties are not met, proper exceptions are returned.
+    """
+    # TODO: verify if regex matches text in a one way only
+
+    # regex must match a whole text from its beginning to end.
+    whole_text_regex = "^" + regex + "$"
+    pattern = re.compile(whole_text_regex)
+
+    match = re.match(pattern, text)
```

Comment:
`object.match(text)` is preferable to passing a compiled regex object as first argument to `re.match()`, so either
```py
matcher = re.match('^%s$' % (regex,), text)
```
or
```py
pattern = re.compile('^%s$' % (whole_text_regex,))
matcher = pattern.match(text)
```
Also, "matcher" is called as such by convention, so please avoid calling it differently if possible.